### PR TITLE
Adds hammer of light

### PR DESCRIPTION
**Pull request by Coxswain-Navigator**
## About The Pull Request
Adds hammer of light, a new ZAYIN

- When the facility is threatened enough, either due to an excess of breached abnormalities or dead players, the seal comes undone.
- When the seal is broken, any agent over level 1 can pick up the hammer to become an incredibly powerful being for (level * 30 seconds). When the time is over, the agent turns into dust.
- The agent in question is armed with an incredibly powerful weapon (the abnormality itself) which deals armor piercing damage and 5% of the target's HP(hard-capped at 500), but does not work on humans.
- Picking up the hammer is entirely optional and unrelated to work, which is rather even across the board. However, the max PE is lower than other zayins at 8 instead of 10.
- Due to a lack of an automated warning system, what the abnormality deems as "threatened enough" is determined by a points-based calculation. Breached abnormalities, non-observing, non-afk ghosts, and running ordeals count towards this. Every time the hammer is used, the required amount of "points" is increased significantly.
- Also ports a visual effect from vgstation.

## Why It's Good For The Game
It isn't.

## Changelog
:cl:
add: Added hammer of light
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/abnormality/zayin/hammer_light.dm [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+		return
+	var/mob/living/carbon/human/H = petter
+	if(!hammer_present)
+		to_chat(H, "<span class='warning'>The hammer is not there!</span>")
+		return
+	if(sealed)
+		to_chat(H, "<span class='warning'>The hammer is sealed!</span>")
+		return
+	if(get_user_level(H) <= 1)
+		to_chat(H, "<span class='warning'>Your body is reduced to atoms by the power of [src]!</span>")
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/zayin/hammer_light.dm:122]: oh fuck yes newest thing to due as a clerk

File: code/modules/mob/living/simple_animal/abnormality/zayin/hammer_light.dm [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+	if(!CanUseEgo(user))
+		return
+	var/statbonus
+	for(var/attribute in user.attributes)
+		var/attribute_level = get_raw_level(user, attribute)
+		statbonus += attribute_level
+	var/damage_multiplier = (clamp(statbonus * 0.25,0, 130) * 0.01 + 1)
+	var/damage_mod = (40 * damage_multiplier) //A damage bonus of up to 130% at max stats, multiplied again by justice
+	var/damage_bonus = clamp(target.maxHealth * 0.05,0, 500)
+	force += (damage_mod + damage_bonus)
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/zayin/hammer_light.dm:224]:
```suggestion
	var/damage_multiplier = (clamp(statbonus * 0.25,0, 130) * 0.01)
	var/damage_mod = (40 * damage_multiplier) //A damage bonus of up to 130% at max stats, multiplied again by justice
	var/damage_bonus = clamp(target.maxHealth * 0.05,0, 500)
	force += (damage_mod + damage_bonus)
```
or
```suggestion
	var/damage_multiplier = (clamp(statbonus * 0.25,0, 130) * 0.01 + 1)
	var/damage_mod = (40 * damage_multiplier) //A damage bonus of up to 130% at max stats, multiplied again by justice
	var/damage_bonus = clamp(target.maxHealth * 0.05,0, 500)
	force = (damage_mod + damage_bonus)
```

File: code/modules/mob/living/simple_animal/abnormality/zayin/hammer_light.dm [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+	pet_bonus = "hums" //saves a few lines of code by allowing funpet() to be called by attack_hand()
+	var/sealed = TRUE
+	var/hammer_present = TRUE
+	var/list/spawned_mobs = list()
+	var/list/banned = list()
+	var/mob/living/carbon/human/current_user = null
+	var/obj/item/ego_weapon/chosen_arms = null
+	var/points
+	var/points_threshold = 150
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/zayin/hammer_light.dm:40]: I don't believe anything increases this, despite what is written in PR

File: code/modules/mob/living/simple_animal/abnormality/zayin/hammer_light.dm [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+/mob/living/simple_animal/hostile/abnormality/hammer_light/proc/Check() //A lot going on here, but basically we assess how bad the situation in the facility is
+	if(!hammer_present)
+		return
+	points = 0
+	for(var/mob/living/simple_animal/hostile/abnormality/A in GLOB.abnormality_mob_list) //How many breaching abnormalities? How dangerous are they?
+		if(A.IsContained())
+			continue
+		switch(A.threat_level)
+			if(ZAYIN_LEVEL)
+				points += 5 //practically nothing
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/zayin/hammer_light.dm:64]: Toad:

**Review by LanceSmites328 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/abnormality/zayin/hammer_light.dm (hunk 3 above)
Comment (Coxswain-Navigator) [reply code/modules/mob/living/simple_animal/abnormality/zayin/hammer_light.dm:40]: addressed in line 134

File: code/modules/mob/living/simple_animal/abnormality/zayin/hammer_light.dm [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+		return
+	PickUpHammer(H)
+	return
+
+//User-related Code
+/mob/living/simple_animal/hostile/abnormality/hammer_light/proc/PickUpHammer(mob/living/carbon/human/user)
+	if(user.ckey in banned)
+		to_chat(user, "<span class='warning'>[src] rejects your cowardice!</span>")
+		user.dust()
+		return
```
[the remark this answers is not in the document]
Comment (Coxswain-Navigator) [reply code/modules/mob/living/simple_animal/abnormality/zayin/hammer_light.dm:133]: That's a fair assessment.

**Review by LanceSmites328 — APPROVED** [review]

**Review by LanceSmites328 — APPROVED** [review]

**Pull request merged**